### PR TITLE
MET-79: Add InternalApi endpoint for creating measurement families

### PR DESCRIPTION
**Pull request by tseho**
**Description (for Contributor and Core Developer)**

New endpoint for the internal API for creating a new MeasurementFamily:
`POST /rest/measurement-families`

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | yes
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

### CODE REVIEW

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        $this->saveMeasurementFamilyHandler = $saveMeasurementFamilyHandler;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        if (!$request->isXmlHttpRequest()) {
+            return new RedirectResponse('/');
+        }
+
+        $normalizedRequest = $this->normalizeRequest($request);
```
Comment (juliensnz) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php]:
```suggestion
        $normalizedRequest = $this->decodeRequest($request);
```

I know you copy and pasted this from asset/RE but this is the correct naming.

FYI: 
JSON string -> [decode] -> php array -> [denormalize] -> php object -> [normalize] -> php array -> [encode] -> JSON string

[decode] + [denormalize] = deserialize
[normalize] + [encode] = serialize 

:)

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/JsonSchema/MeasurementFamilyStructureValidator.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        $normalizedMeasurementFamilyObject = Validator::arrayToObjectRecursive($normalizedMeasurementFamily);
+        $validator->validate($normalizedMeasurementFamilyObject, $this->getJsonSchema());
+
+        return $validator->getErrors();
+    }
+
+    private function getJsonSchema(): array
+    {
+        return [
+            'type' => 'object',
```
Comment (juliensnz) [inline src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/JsonSchema/MeasurementFamilyStructureValidator.php]: couldn't we store this as json and use the same for back and front?

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php (hunk 1 above)
Comment (juliensnz) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php]: cc @SamirBoulil @ValentinMumble :)

Comment (tseho) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php]: It's not even from asset/RE, I've kind of followed the ExternalApi from the MeasureBundle: https://github.com/akeneo/pim-community-dev/blob/cb0041544c45c9e74a92d5f49a4d514848b4c391/src/Akeneo/Tool/Bundle/MeasureBundle/Controller/ExternalApi/SaveMeasurementFamiliesAction.php#L94

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/JsonSchema/MeasurementFamilyStructureValidator.php (hunk 2 above)
Comment (tseho) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/JsonSchema/MeasurementFamilyStructureValidator.php]: Yes, we should need it in the front. It's also copied from `MeasureBundle/Controller/ExternalApi`, I don't know if we will have issues if this is shared between those.

File: src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php (hunk 1 above)
Comment (tseho) [reply src/Akeneo/Tool/Bundle/MeasureBundle/Controller/InternalApi/CreateMeasurementFamilyAction.php]: done

**Pull request merged**